### ISSUE TITLE
fakeServer uses FakeXDomainRequest instead of FakeXMLHttpRequest in node

### ISSUE THREAD

**Issue by berniegp** [sinonjs/sinon#1163]
- Sinon version : 1.17.6
- Environment   : node v6.7.0

Using `sinon.fakeServer.create()` does not produce the same result as using `sinon.useFakeXMLHttpRequest()` when mocking `XMLHttpRequest`. The latter replaces `XMLHttpRequest` with `FakeXMLHttpRequest` whereas the former uses `FakeXDomainRequest`.

This is a problem because `FakeXMLHttpRequest` has an `upload` property of type `UploadProgress` while `FakeXDomainRequest` does not. That property is used to attach upload progress event listeners to an `XMLHttpRequest`.

Based on my comprehension of the documentation, I would expect fakeServer to have the same behavior as useFakeXMLHttpRequest and use `FakeXMLHttpRequest`. Also, I understand that `XDomainRequest` is actually [non-standard and deprecated](https://developer.mozilla.org/en-US/docs/Web/API/XDomainRequest) so perhaps it should be explicitly requested when desired by the user.

This behavior was introduced by #584 for an IE9 issue.

lib/sinon/util/fake_server.js defines this behavior (lines 71-87):

```
sinon.fakeServer = {
    create: function (config) {
        var server = sinon.create(this);
        server.configure(config);
        if (!sinon.xhr.supportsCORS) {
            this.xhr = sinon.useFakeXDomainRequest();  // <------- here!
        } else {
            this.xhr = sinon.useFakeXMLHttpRequest();
        }
        server.requests = [];

        this.xhr.onCreate = function (xhrObj) {
            server.addRequest(xhrObj);
        };

        return server;
    },
```

Running my test with node (i.e. not in a browser) reports that `sinon.xhr.supportsCORS` is false which triggers this behavior.

**Comment by berniegp** [comment]
https://github.com/sinonjs/sinon/issues/732#issuecomment-252412559 provides a workaround:

```
sinon.useFakeXDomainRequest = sinon.useFakeXMLHttpRequest;
```

**Comment by mroderick** [comment]
How did you get this far in node, considering that there is no native XMLHttpRequest API?

Can we see some code and test that fails?

Keeping in mind that FakeXMLHttpRequest and FakeServer were never intended to work in Node.JS, how do you propose we solve this issue?

**Comment by berniegp** [comment]
My code is not executed in node in the end. It is destined for client-side (browser) use which I test in node and pack with Browserify. I find it easier to test my js code in node.

Eventually, I gathered as much while hacking on FakeServer and co that they were not really meant for use in node. I actually ended up rolling my own [XMLHttpRequest mock](https://github.com/berniegp/mock-xmlhttprequest) based on the spec without any browser feature support check. So I guess this issue might be out of scope for sinon.
